third_party/ienn/src/ie_compilation.cpp

```diff
@@ -359,155 +316,166 @@ int32_t Compilation::AddConvolution(const Operation& operation) {
-            .setStrides({params.stride_width, params.stride_height})
-            .setPaddingsBegin({params.padding_top, params.padding_left})
-            .setPaddingsEnd({params.padding_bottom, params.padding_right})
-            .setPoolingType(operation.type == AVERAGE_POOL_2D
-                                ? Builder::PoolingLayer::PoolingType::AVG
-                                : Builder::PoolingLayer::PoolingType::MAX)
-            .setRoundingType(Builder::PoolingLayer::RoundingType::FLOOR));
-    if (params.fuse_code != FUSED_NONE) {
-      result = AddActivationByFusedCode(params.fuse_code, layer_id, output_name,
-                                        layer_id);
+      if (params.output_channel !=
+          params.input_channel * params.depthwise_multiplier) {
+        std::cout
+            << "Failed assertion: outputFeatureChannels "
+            << params.output_channel
+            << " in IE depthwise convolution descriptor must be multiplie of "
+               "inFeatureChannels "
+            << params.input_channel;
+        return error_t::BAD_DATA;
+      }
+      Blob::Ptr blob;
+      result = CreateBlob(weights_index, blob);
       if (result != error_t::NOT_ERROR) {
         return result;
       }
+      SizeVector weights_dims{params.depth_out, 1, 1, params.filter_height,
+                              params.filter_width};
+      auto weights_node = std::make_shared<op::Constant>(
+          element::f32, Shape(weights_dims), blob->buffer().as<float*>());
+      conv_node = std::make_shared<op::v1::GroupConvolution>(
+          index_op_map_[input_index], weights_node->output(0), Strides(strides),
+          CoordinateDiff(padBegin), CoordinateDiff(padEnd), Strides(dilations));
+    } else {
+      AddConstant(weights_index);
+      conv_node = std::make_shared<op::v1::Convolution>(
+          index_op_map_[input_index], index_op_map_[weights_index],
+          Strides(strides), CoordinateDiff(padBegin), CoordinateDiff(padEnd),
+          Strides(dilations));
     }
-    layer_id_map_[output_index] = layer_id;
-  } catch (const std::exception& ex) {
-    std::cout << "[IE] failed to add pooling layer " << ex.what();
-    return error_t::OP_FAILED;
-  }
-  return error_t::NOT_ERROR;
-}
-
-int32_t Compilation::AddSoftmax(const Operation& operation) {
-  // Setup softmax params.
-  SoftmaxParams params;
-  int32_t result = GetSoftmaxParams(model_, operation, params);
-  if (result != error_t::NOT_ERROR)
-    return result;
-  // Check beta.
-  if (params.beta != 1.0) {
-    std::cout << "beta " << params.beta << " is not supported.";
-    return error_t::BAD_DATA;
-  }
-  const uint32_t input_index = operation.inputs[0];
-  if (layer_id_map_.find(input_index) == layer_id_map_.end()) {
-    std::cout << "The layer for operand index " << input_index
-              << " is not ready";
-    return error_t::BAD_DATA;
-  }
-  try {
+    // add bias and fused-code
+    AddConstant(bias_index);
```

Comment:
How about moving the section of boradcasting bias to a funtion so that we can read code more easy.